webrender/src/device/gl.rs

```diff
@@ -1232,7 +1232,7 @@ impl Device {
                     0,
                     desc.external,
                     desc.pixel_type,
-                    pixels.map(texels_to_u8_slice),
+                    pixels.and_then(texels_to_u8_slice),
```

Comment:
I don't think this is correct. If the slice is provided, we exactly X bytes of data. Just ignoring empty data slices and treating them as storage allocation without initialization is something we shouldn't need to do.